04 - Poker.playground/Contents.swift

```diff
@@ -5,16 +5,145 @@
+        case .Kreuz: return "\u{2663}"
+        }
+    }
+}
+
+enum Ranking: Int {
+    case straightflush = 1, fourofakind = 2, fullhouse = 3, flush = 4, straight = 5, threeofakind = 6, twopair = 7, onepair = 8, highcard = 9
+    var description: String{
+        switch self{
+        case .straightflush: return "Straight flush"
+        case .fourofakind: return "Four of a kind"
+        case .fullhouse: return "Full house"
+        case .flush: return "Flush"
+        case .straight: return "Straight"
+        case .threeofakind: return "Three of a kind"
+        case .twopair: return "Two pair"
+        case .onepair: return "One pair"
+        case .highcard: return "High card"
+        }
+    }
+}
+
+/*Wieso struct? Weil Karten statische Dinge sind*/
+struct Card: CustomStringConvertible {
+    /*Wieso enum? Weil sich die Faelle gegenseitig ausschließen und nur eine Karte eine Farbe und einen Rang haben kann */
+    
+    let suit: Suit
+    let rank: Rank
+    
+    var description: String{
+    return "\(suit) \(rank)"
+    }
+    
+}
+
+extension Card: Equatable{}
+func ==(lhs: Card, rhs: Card) -> Bool {
+    return lhs.suit == rhs.suit && lhs.rank == rhs.rank
+}
+
+struct PokerHand{
+    let cards: [Card]
+    var ranking: Ranking
+    
+    init(){
+        var p1 = Card(suit: Suit(rawValue: 1)!, rank: Rank(rawValue: 1)!)
+        var p2 = Card(suit: Suit(rawValue: 1)!, rank: Rank(rawValue: 1)!)
+        var p3 = Card(suit: Suit(rawValue: 1)!, rank: Rank(rawValue: 1)!)
+        var p4 = Card(suit: Suit(rawValue: 1)!, rank: Rank(rawValue: 1)!)
+        var p5 = Card(suit: Suit(rawValue: 1)!, rank: Rank(rawValue: 1)!)
+        while p1==p2 || p1==p3 || p1==p4 || p1==p5 || p2==p3 || p2==p4 || p2==p5 || p3==p4 || p3==p5 || p4==p5{
+            var rndSuit = Suit(rawValue: Int(arc4random_uniform(4)+1))!
+            var rndRank = Rank(rawValue: Int(arc4random_uniform(13)+1))!
+            var rndCard = Card(suit: rndSuit, rank: rndRank)
+            p1 = rndCard
+            rndSuit = Suit(rawValue: Int(arc4random_uniform(4)+1))!
+            rndRank = Rank(rawValue: Int(arc4random_uniform(13)+1))!
+            rndCard = Card(suit: rndSuit, rank: rndRank)
+            p2 = rndCard
+            rndSuit = Suit(rawValue: Int(arc4random_uniform(4)+1))!
+            rndRank = Rank(rawValue: Int(arc4random_uniform(13)+1))!
+            rndCard = Card(suit: rndSuit, rank: rndRank)
+            p3 = rndCard
+            rndSuit = Suit(rawValue: Int(arc4random_uniform(4)+1))!
+            rndRank = Rank(rawValue: Int(arc4random_uniform(13)+1))!
+            rndCard = Card(suit: rndSuit, rank: rndRank)
+            p4 = rndCard
+            rndSuit = Suit(rawValue: Int(arc4random_uniform(4)+1))!
+            rndRank = Rank(rawValue: Int(arc4random_uniform(13)+1))!
+            rndCard = Card(suit: rndSuit, rank: rndRank)
+            p5 = rndCard
+        }
+        cards = [p1, p2, p3, p4, p5]
+        ranking = Ranking(rawValue: 9)!
```

Comment:
Funktioniert, aber kannst du auch eleganter schreiben ;) Versuch's mit einem Array `var cards: [Card] = []` vor `while cards.count < 5` und dann in der Schleife eine neue zufällige Karte erstellen und nur `cards` hinzufügen wenn `!cards.contains(card)`.